sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala

```diff
@@ -200,6 +200,15 @@ object ScalaJSPlugin extends AutoPlugin {
         "Entry points of the Scala.js application, to be called when it starts.",
         AMinusTask)
 
+    val scalaJSUseMainEntryPoint = SettingKey[Boolean]("scalaJSUseMainEntryPoint",
+        "If true, adds the `mainClass` as an entry point of the Scala.js module",
+        APlusSetting)
+
+    val scalaJSMainEntryPoint = TaskKey[Option[EntryPoint]](
+        "scalaJSMainEntryPoint",
+        "The main entry point, derived from `mainClass` if `scalaJSUseMainEntryPoint` is true",
+        BPlusTask)
```

Comment:
If their main class is not a `js.JSApp`, or in general if their `EntryPoint` doesn't use `"main"` as the main method name.